### PR TITLE
Add test coverage for async slicing of points and fix rounding bug

### PR DESCRIPTION
**Pull request by andy-sweet**
# Description

This PR adds some async slicing integration tests for points, one of which was failing due to unintended rounding. It also adds some debug logs across sync and async slicing paths to make it easier to debug issues. Finally, it changes the main log message format to include the thread name, which is incredibly helpful when debugging concurrency issues.

# References

This was split off from #5711 to avoid bloating that already large PR. Credit to @kcpevey, who found the the bug and wrote most of #5711. Also this part of the larger #4795

## Type of change
- [x] Bug-fix (non-breaking change which fixes an issue)

# How has this been tested?
- [x] added new tests to cover issues
- [x] all existing tests pass with my change

I manually tested behavior and logging with async off and on.

### CODE REVIEW

File: napari/_qt/_tests/test_async_slicing.py [hunk 1]
```diff
@@ -129,3 +174,17 @@ def assert_vispy_image_data_equal() -> None:
...
+
+def wait_until_vispy_points_data_equal(
+    qtbot, vispy_layer: VispyImageLayer, expected_data: np.ndarray
+) -> None:
+    def assert_vispy_points_data_equal() -> None:
+        positions = vispy_layer.node._subvisuals[0]._data['a_position']
+        # Flip the coordinates because vispy uses xy instead of rc ordering.
+        # Also only take the number of dimensions expected since vispy points
+        # are always 3D even when displaying 2D slices.
+        data = positions[:, -expected_data.shape[1] :: -1]
```
Comment (kcpevey) [inline napari/_qt/_tests/test_async_slicing.py:187]: its these nuances that make coding in new areas so challenging for me (or other noobs). Thanks for the comment here!

**Review by kcpevey — APPROVED** [review]
This looks great. And added logging is a definitive bonus.

**Comment by Czaki** [comment]
Is it intentional to use random generator without setting rng state?

**Comment by andy-sweet** [comment]
> Is it intentional to use random generator without setting rng state?

My intent was to use the existing `rng` fixture to provide a generator seeded with 0: https://github.com/napari/napari/pull/5783/files#diff-35dd4aa64ecbf154386ef80c30c2dc8c80b55b8a6cb35cc6749df2bcfcee407dR18

That way I can avoid accidentally using unseeded random numbers and also avoid the verbose two liner `np.random.seed(0); np.random.rand(...)`.

But I'm still fairly new to using the new numpy style of randoms - did I misunderstand something?

**Comment by andy-sweet** [comment]
@kcpevey , @jni : I considered bringing `Layer.loaded` into the mix here, but I decided to do that in a separate PR where we can have a more dedicated discussion around it. I think these changes are independently useful and easier to review on their own.

**Comment by brisvag** [comment]
Reminder for myself that currently dims is a bit broken and these tests will probably need an update once we get #5751 in.

**Comment by jni** [comment]
Still catching up after my 1w break but please ping me on Zulip if I fail to review this in the coming days and it remains stuck...! (And #5751)

**Comment by andy-sweet** [comment]
@jni : added you as a reviewer since you took a quick look at this in the architecture working group and seemed fine with it

**Review by brisvag — APPROVED** [review]
LGTM!

**Comment by andy-sweet** [comment]
Thanks for the review! I'll likely merge this on Monday, but if anyone wants to beat me to it, feel free :)

**Pull request merged**